### ISSUE TITLE
Clear cached data on print end

### ISSUE THREAD

**Issue by StefanCohen** [j7126/OctoPrint-Dashboard#177]
The accumulated stats should be cleared on print success/failure/cancel so the widgets are  cleared on the next print start.

**Comment by fixyourlan** [comment]
This should be cleared when a new print is started, not at the end, so you can still see the stats of the last print, as it is now its cleared as soon as its finished and you are unable to see how long it took to print. Before it was just leaving the old info even after the new print was started, I think it was until the first layer was completed, which was confusing, now its cleared too soon for me!

**Comment by StefanCohen** [comment]
@j7126 @Willmac16 The comment from @fixyourlan is interesting as it in conflict with my original design vision. See #145 for context. I created this ticket to go in the other direction and clear everything on print end/cancel. I mentioned this on our call earlier today but I think this puts it into context a bit better.

**Comment by j7126** [comment]
I think maybe it could be a setting to choose if it gets cleared at the start or the end for each widget. Additionally that setting would control if that widget is still shown when no print is running, this way for example you could still see they layer times graph  form the previous print after it is done (which currently even though it is cleared at print start, it is not visible until print start anyway).

**Comment by j7126** [comment]
As per my previous comment, I have implemented a setting for each widget which controls if a widget is hidden when not currently printing or not. 

I have implemented a setting that controls if the data is cleared on print start, print end or never for the printer message, layer graph, filament, and feed-rate widgets. 

These settings are located in the new visibility settings modal for each widget, this modal also contains the 'show in full-screen' checkbox, and the button for this modal is located where that checkbox was.
